### PR TITLE
Adds the Automated Announcement System

### PR DESCRIPTION
**Pull request by Chiefwaffles**
The Automated Announcement System is a new structure (currently placed in TComms control room) that now handles two functions that would otherwise be handled by available AIs:
- Latejoin Announcement
- Departmental Head announcements (see #10463 for the AI version of this)

Now instead of the game choosing an AI to state things such as "Person McName has signed up as Captain", the game chooses [an/the] AAS to do so. This means that even without an AI, people can be still notified of new arrivals and new heads. The game should support multiple AASes just in case, though there's no need to have more than one. 
Keep in mind that with this change, **maps without an AAS will no longer have latejoin announcements.** I can add legacy support and make the game look for an AI if it can't find an AAS if needed.

Further more, regular players (or the AI) are able to customize the messages to their liking, using the phrases "%PERSON" and "%RANK" to signify the locations of the variables. Anyone with access to TComms or the AI will be able to change the messages to just about anything. It can also be broken by EMPs/emags and will spout out useless messages on their appropriate event instead of one with actual information.

If you don't feel like reading #10463, the gist is that when a head joins the game at either roundstart or via latejoining, the AAS will announce their presence to their department channels. For instance, if the RD joins the game, the AAS will say in the science channel "Person McScience, Research Director, is the new department head."

And of course, remember that I'm new to byond coding and that my solutions to problems may be horrifying. I'd appreciate it if you pointed out those things as you see them.

### CODE REVIEW

**Comment by Dorsisdwarf** [comment]
I don't like this. The arrivals announcement has always been the go-to "do we have an AI" check.

**Comment by WJohn** [comment]
Frankly that could be fixed in the same way that we have a message saying "Captainship not forced on anyone", except for the AI.

**Comment by bustygoku** [comment]
I like that the AI is forced to bark arrivals announcements personally.

**Comment by RandomMarine** [comment]
Or just check your PDA messenger and see if there's an AI on the list.

This also fixes the unreported extreme edge case issue of a shunted malf AIs possibly giving itself away with a forced "Rockchest McBumblehead has signed up as Assistant." message.

**Comment by Miauw62** [comment]
Make it play voxwelcome automatically only if there is an AI?
Op 13-jul.-2015 13:16 schreef "bustygoku" <notifications@github.com>:

File: code/game/jobs/job/job.dm [hunk 1]
```diff
@@ -171,4 +171,10 @@
...
 /datum/job/proc/config_check()
-	return 1
\ No newline at end of file
+	return 1
+
+/datum/job/proc/announce_head(var/datum/mind/o_mind, var/channels) //tells the given channel that the given mind is the new department head. See communications.dm for valid channels.
+	sleep(2) //to allow some initialization
+	if(announcement_systems.len)
+		var/obj/structure/announcement_system/announcer = pick(announcement_systems)
+		announcer.announce("[o_mind.name], [o_mind.assigned_role], is the new department head.", channels)
```
Comment (RemieRichards) [inline code/game/jobs/job/job.dm]: "new"? seems needless to me.

File: code/game/objects/structures/announcement_system.dm [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	icon = 'icons/obj/radio.dmi'
+	icon_state = "beacon"
+	var/obj/item/device/radio/headset/radio
+
+/obj/structure/announcement_system/New()
+	announcement_systems += src
+	radio = new /obj/item/device/radio/headset/ai(src)
+
+/obj/structure/announcement_system/proc/announce(var/message, list/channels)
+	spawn(1) //to wait for people joining, roundstart init, etc.
```
Comment (RemieRichards) [inline code/game/objects/structures/announcement_system.dm]: this spawn is empty so it's actually not doing anything, and can be removed.

**Comment by Iamgoofball** [comment]
@Dorsisdwarf Maybe try checking your PDA?

I think this is pretty damn cool.

If you're up for it, you should add a feature that allows people to fuck with it and put in absolutely bogus dumb messages when people join, such as replacing the "[NAME] has signed up as [JOB]" with "HERE IN THE SPACE WWE SUUUUUPER SLAMMMMMM [NAME] IS ENTERING THE RING OF [JOB] FIGHTERS".

It'd be fun. Ignore anyone who tells you this is useless, they're a nerd who hates fun stuff.

**Comment by GunHog** [comment]
Is the AI able to program it remotely, as with all T-comms related equipment? I have been toying with the auto-announcement message as AI using NTSL for months, it is not so bad to have another machine do it, so long as it can be configured. Even so, until the day that NTSL is replaced, the message can still be customized via radio!

**Comment by Chiefwaffles** [comment]
Alright, AIs and people with multitools can now configure the message. Code isn't as nice now, but it still works fine. This should be the final change.
ss13 screen3
(ignore "departmend")

**Comment by GunHog** [comment]
I would go with "a Head of Staff" rather than "department head".

**Comment by RandomMarine** [comment]
If it's announced to only the department's radio channel, it should say something along the lines of "Your department head %PERSON has arrived on board."

**Comment by Chiefwaffles** [comment]
It's only announced to the head's respective radio channels, GunHog, so "departmend head" should work just fine in the context of that one channel.
The "Your department head %PERSON has arrived on board" message is a little problematic because it'd be a biit strange at roundstart, and I find it'd be best to include the job for newbies/edge cases like the HoP's presence being announced to cargo.

**Comment by Alek2ander** [comment]
Feature request: toggle both types of announcements on and off
Feature request: button to change default radio channel

**Comment by zylowalsh** [comment]
It would be cool if it left a message on the sec channel when one of the cell timers are up.

**Comment by GunHog** [comment]
>It would be cool if it left a message on the sec channel when one of the cell timers are up.

This is already the case! Cell timers transmit on the radio, as well as Securitrons.

**Comment by Chiefwaffles** [comment]
IT'S
DONE

ss13 screen5

You can now toggle channels on/off, emag/emp the AAS, construct more/deconstruct them, repair them after an emag/emp, and admire the cool sprite thanks to ZyloWalsh.

**Comment by Chiefwaffles** [comment]
Can someone go over the code and check for bad segments? I'd really appreciate it, as I'm more or less done (unless something else comes up) with it now.

**Comment by GunHog** [comment]
You have a merge conflict. Update your master to the current code, merge it with this branch, and resolve accordingly. (If you have never done this before, seek help in coderbus before you try - you could damage your PR if you are not careful)

**Comment by Chiefwaffles** [comment]
Huh. Didn't notice that.
Though it should be working now.

**Comment by Chiefwaffles** [comment]
Okay. I should have fixed the second conflict and semi-fixed my mess-up involving fixing the second conflict.
Though git thinks I touched the malfunction.dm file now, even though I didn't.

**Comment by OhEmGee** [comment]
>[NAME] has signed up as [JOB]" with "HERE IN THE SPACE WWE SUUUUUPER SLAMMMMMM [NAME] IS ENTERING THE RING OF [JOB] FIGHTERS".

ALL OF MY YES

**Comment by Iamgoofball** [comment]
@Chiefwaffles Can you fix the conflicts on this?

**Comment by Chiefwaffles** [comment]
Yeah. I plan on doing that soon. It's getting a bit tedious fixing the unrelenting conflicts though.

**Comment by Chiefwaffles** [comment]
Okay. It should work now. Even though I didn't even touch the conflicting files.

Hmph.

**Comment by WJohn** [comment]
:-1: on the sprites. I'll look into making them better at a later date after this gets merged. If it's not merged tonight or early tomorrow I'll go ahead and fix those up to look much better.

**Comment by WJohn** [comment]
As promised: http://www.mediafire.com/download/5ddqd6i4d1lwj2m/Sprites_Request_11.0.5b.rar
It's in other.dmi

Also why is the sprite in its own .dmi? You could just group it up with the other telecomms machinery since they all have to do with communication.

**Comment by Chiefwaffles** [comment]
Alright. I'll add in the new sprites into the telecomms dmi (was unsure before because the AAS isn't used for regular telecomms operation like the other machines) as well as fix THE FIFTH OR SO MERGE CONFLICT RRRGH.

**Comment by optimumtact** [comment]
@Cheridan @RemieRichards @Miauw62 @phil235 @Aranclanos @Jordie0608 

Come on guys, poor dude has had to fix merge conflicts like twice

File: code/game/jobs/job/job.dm [hunk 3]
```diff
@@ -167,4 +167,10 @@
 	return max(0, minimal_player_age - C.player_age)
 
 /datum/job/proc/config_check()
-	return 1
\ No newline at end of file
+	return 1
+
+/datum/job/proc/announce_head(var/datum/mind/o_mind, var/channels) //tells the given channel that the given mind is the new department head. See communications.dm for valid channels.
+	spawn(4) //to allow some initialization
```
Comment (optimumtact) [inline code/game/jobs/job/job.dm]: What on earth is this for?

**Comment by CorruptComputer** [comment]
When this is merged I have an update planned for AsteroidStation.

**Pull request merged**